services/QuillLMS/app/controllers/auth/classlink_controller.rb

```diff
@@ -1,37 +1,30 @@
 # frozen_string_literal: true
 
-require 'net/http'
-require 'uri'
-
 module Auth
   class ClasslinkController < ApplicationController
+    class ClasslinkUnsuccessfulLoginError < StandardError; end
     around_action :force_writer_db_role, only: [:classlink]
 
-    def classlink
-      code = params['code']
-
-      uri = URI.parse(::Classlink::OAUTH_TOKEN_ENDPOINT)
-      form_data = {
-        client_id: ::Classlink::CLIENT_ID,
-        client_secret: ::Classlink::CLIENT_SECRET,
-        code:
-      }
-
-      response = Net::HTTP.post_form(uri, form_data)
-      parsed = JSON.parse response.body
+    attr_accessor :code
 
-      # TODO: everything below here was just for proof of concept purposes.
-      # Delete before public launch
-      # uri = URI('https://nodeapi.classlink.com/v2/my/info')
-      # http = Net::HTTP.new(uri.host, uri.port)
-      # http.use_ssl = true
-      # request = Net::HTTP::Get.new(uri.request_uri)
-
-      # request['Authorization'] = "Bearer #{parsed['access_token']}"
+    def classlink
+      @code = params[:code]
+
+      begin
+        sign_in(user)
+        redirect_to profile_path
+      rescue => e
+        ErrorNotifier.report(ClasslinkUnsuccessfulLoginError.new)
+        redirect_to root_path, flash: { error: 'Classlink log in unsuccessful' }
```

Comment:
Not part of the spec, but I think it's a nice improvement, rather than a 404 page 

<img width="1777" alt="Screenshot 2025-04-04 at 4 00 45 PM" src="https://github.com/user-attachments/assets/88b52513-60fa-4db3-b276-1ec4becb4550" />